vpr/src/place/place_constraints.cpp

```diff
@@ -36,7 +36,7 @@ bool is_cluster_constrained(ClusterBlockId blk_id) {
     return (!pr.empty());
 }
 
-bool is_macro_constrained(t_pl_macro pl_macro) {
+bool is_macro_constrained(t_pl_macro& pl_macro) {
```

Comment:
for all the things being passed by reference, make it a const reference to indicate that you are doing this for efficiency

Reply:
also better for the compiler to have it const if you are not changing it